.github/workflows/preview.yml

```
@@ -96,12 +98,23 @@ jobs:
         run: |
           echo "The PR preview for ${{ fromJSON(env.PR_DATA).head_sha }} is available at [${{ env.PREVIEW_DOMAIN }}](https://${{ env.PREVIEW_DOMAIN }})" >> pr.md
           echo "" >> pr.md
-          echo "The following output files are affected by this PR:" >> pr.md
-          sed -E "s#(.*)#- [\1](https://${{ env.PREVIEW_DOMAIN }}/${{ fromJSON(env.PR_DATA).target_name }}/\1)#" diff.txt >> pr.md
-          echo "" >> pr.md
-          echo '[show diff](https://${{ env.PREVIEW_DOMAIN }}/diff.patch)' >> pr.md
-          echo "" >> pr.md
-          echo '[show diff as HTML](https://${{ env.PREVIEW_DOMAIN }}/diff.html)' >> pr.md
+
+          if [[ -f diff.txt ]] ; then
```

```suggestion
          if [ -f diff.txt ] ; then
```
AFAIK, `[[ ]]` is for conditional command execution, whereas `[ ]` is `test`, which is probably what you're trying to do.
If you're using `[[ ]]` with an intent, can you please explain why is it better here than `[ ]` (`test`)?

---

`[[` is a bash version of `[`. https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html explains some of the differences. In this particular case it probably doesn't matter but it absolutely can:
```sh
#!/bin/bash

if [ -n $DOESNOTEXIST ] ; then
	echo "plain: The length of '$DOESNOTEXIST' is nonzero"
else
	echo "plain: The length of '$DOESNOTEXIST' is zero"
fi

if [[ -n $DOESNOTEXIST ]] ; then
	echo "bash: The length of '$DOESNOTEXIST' is nonzero"
else
	echo "bash: The length of '$DOESNOTEXIST' is zero"
fi

VALUEWITHSPACES="this has spaces"
if [ -n $VALUEWITHSPACES ] ; then
	echo "plain: The length of '$VALUEWITHSPACES' is nonzero"
else
	echo "plain: The length of '$VALUEWITHSPACES' is zero"
fi

if [[ -n $VALUEWITHSPACES ]] ; then
	echo "bash: The length of '$VALUEWITHSPACES' is nonzero"
else
	echo "bash: The length of '$VALUEWITHSPACES' is zero"
fi
```

When you run it:
```console
$ bash example.sh
plain: The length of '' is nonzero
bash: The length of '' is zero
example.sh: line 16: [: too many arguments
plain: The length of 'this has spaces' is zero
bash: The length of 'this has spaces' is nonzero
```

You can see the plain check does not what you expect it to do and actually crashes at some point. Now there are POSIX shell compliant options to make it work and [shellcheck](https://www.shellcheck.net/) will loudly yell at you about these, but in general if you can assume bash it's easier to use `[[`.

---

Thank you.